lib/ansible/modules/source_control/git.py

```diff
@@ -371,9 +371,10 @@ def write_ssh_wrapper():
 
 def set_git_ssh(ssh_wrapper, key_file, ssh_opts):
 
-    if os.environ.get("GIT_SSH"):
-        del os.environ["GIT_SSH"]
+    # git_ssh_command will override git_ssh, so only older git needs it
```

Comment:
As far as I see `GIT_SSH_COMMAND` was only introduced in git 2.3, so this change will not fix the issue on many systems, including rhel7.